code/datums/components/beacon.dm

```diff
@@ -128,15 +131,21 @@
 	user.show_message(span_notice("The [source] beeps and states, \"Your current coordinates were registered by the supply console. LONGITUDE [location.x]. LATITUDE [location.y]. Area ID: [get_area(source)]\""), EMOTE_AUDIBLE, span_notice("The [source] vibrates but you can not hear it!"))
 	beacon_datum = new /datum/supply_beacon("[user.name] + [A]", get_turf(source), user.faction)
 	RegisterSignal(beacon_datum, COMSIG_QDELETING, PROC_REF(clean_beacon_datum))
+	RegisterSignal(source, COMSIG_MOVABLE_MOVED, PROC_REF(updatepos))
+	if(source.loc)
```

Comment:
This will also register trash signal to turfs when a regular beacon is put down.